pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -760,6 +761,18 @@ func newCloud(config io.Reader) (*Cloud, error) {
 		AssumeRoleTokenProvider: stscreds.StdinTokenProvider,
 	}))
 
+	// If region has not been provided, let's try to get it from an EC2
+	// metadata service and fail if we cannot get that way.
+	if *sess.Config.Region == "" {
+		s := session.Must(session.NewSession(aws.NewConfig().WithMaxRetries(0)))
+		m := ec2metadata.New(s)
+		r, err := m.Region()
+		if err != nil {
+			return &Cloud{}, errors.New("unable to determine region")
+		}
+		sess.Config.Region = &r
```

Comment:
A user does not need to know if this happens. This is a standard way to initialize `AWS`. A user can set `AWS_DEFAULT_REGION`, etc. I agree, this will be documented in the docs.